seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java

```diff
@@ -0,0 +1,179 @@
+        String tableName = config.getString(KuduSourceConfig.tableName);
+
+        String columnslist = config.getString(KuduSourceConfig.columnsList);
+
+
+        kuduInputFormat=new KuduInputFormat(kudumaster,tableName,columnslist);
+        try {
+            KuduClient.KuduClientBuilder kuduClientBuilder = new
+                    KuduClient.KuduClientBuilder(kudumaster);
+            kuduClientBuilder.defaultOperationTimeoutMs(1800000);
+
+            KuduClient kuduClient = kuduClientBuilder.build();
+            partitionParameter = initPartitionParameter(kuduClient,tableName);
+            SeaTunnelRowType seaTunnelRowType =getSeaTunnelRowType(kuduClient.openTable(tableName).getSchema().getColumns());
+
+
+            rowTypeInfo=seaTunnelRowType;
+        } catch (KuduException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private PartitionParameter initPartitionParameter(KuduClient kuduClient,String tableName) {
+        String keyColumn = null;
+        ArrayList<Integer> keyList = null;
+        try {
+            KuduScanner.KuduScannerBuilder kuduScannerBuilder =
+                    kuduClient.newScannerBuilder(kuduClient.openTable(tableName));
+            ArrayList<String> columnsList = new ArrayList<String>();
+            keyColumn = kuduClient.openTable(tableName).getSchema().getPrimaryKeyColumns().get(0).getName();
+            columnsList.add(""+keyColumn);
+            kuduScannerBuilder.setProjectedColumnNames(columnsList);
+
+            KuduScanner kuduScanner = kuduScannerBuilder.build();
+            keyList = new ArrayList<>();
+            while (kuduScanner.hasMoreRows()) {
+                RowResultIterator rowResults = kuduScanner.nextRows();
+                while (rowResults.hasNext()) {
+                    RowResult row = rowResults.next();
+                    int id = row.getInt(""+keyColumn);
+                    keyList.add(id);
+
+                }
+
+            }
+        } catch (KuduException e) {
+            e.printStackTrace();
+        }
+        Collections.sort(keyList);
+
+        return new PartitionParameter(keyColumn, Long.parseLong(keyList.get(0).toString()), Long.parseLong(keyList.get(keyList.size()-1).toString()));
+    }
+
+
+    @Override
+    public void setSeaTunnelContext(SeaTunnelContext seaTunnelContext) {
+        this.seaTunnelContext = seaTunnelContext;
+    }
+
+
+    public SeaTunnelRowType getSeaTunnelRowType(List<ColumnSchema> columnSchemaList) {
+
+        ArrayList<SeaTunnelDataType<?>> seaTunnelDataTypes = new ArrayList<>();
+        ArrayList<String> fieldNames = new ArrayList<>();
+        try {
+
+            for (int i = 0; i < columnSchemaList.size(); i++) {
+                fieldNames.add(columnSchemaList.get(i).getName());
+                seaTunnelDataTypes.add(KuduTypeMapper.mapping(columnSchemaList, i));
+            }
+
+        } catch (Exception e) {
+            LOGGER.warn("get row type info exception", e);
```

Comment:
If the log is only printed here, this method will return a problematic `SeaTunnelRowType`. I'm not sure whether this will cause problems.